### PR TITLE
Fix cfg; test actual compilation

### PR DESCRIPTION
**Pull request by pb-cdunn**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pb-cdunn** [comment]
Right now, the compiler path is hard-coded, obviously wrong.
```
/bin/sh: /home/conda/feedstock_root/build_artifacts/nim_1573197297071/_build_env/bin/x86_64-conda_cos6-linux-gnu-cc: No such file or directory
```
 But `gcc` doesn't seem to be in `$PATH` for the new test.
```
bin/sh: gcc: command not found
```

@conda-forge/staged-recipes , What's the proper way to find gcc at test/run-time?

**Comment by tadeu** [comment]
The compiler is indeed that one (`x86_64-conda_cos6-linux-gnu-cc`), `conda-build` uses crosstool-NG to ensure that there are no interference from system packages in the compilation, so it's good that no `gcc` where found in `$PATH`.

Take a look here for more info: https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html

File: recipe/build.sh [hunk 1]
```diff
@@ -6,10 +6,10 @@ set -vexu -o pipefail
 
 pushd nim
 # inject compilers
-echo "gcc.exe = \"$CC\"" >> config/nim.cfg
-echo "gcc.linkerexe = \"$CC\"" >> config/nim.cfg
-echo "clang.exe = \"$CC\"" >> config/nim.cfg
-echo "clang.linkerexe = \"$CC\"" >> config/nim.cfg
+echo "gcc.exe = \"$(basename $CC)\"" >> config/nim.cfg
+echo "gcc.linkerexe = \"$(basename $CC)\"" >> config/nim.cfg
```
Comment (mariusvniekerk) [inline recipe/build.sh]: Shoudln't this be LD ?

Comment (pb-cdunn) [reply recipe/build.sh]: I've asked around, and we're pretty sure we want to link using gcc, rather than using ld. Anway, the crash is at compile-time.

**Comment by pb-cdunn** [comment]
That link helped, and we have a couple ideas we're going to try ...

**Comment by pb-cdunn** [comment]
While debugging the Azure build, I noticed that `$PATH` contains a missing directory:

```
2019-11-09T18:11:08.6836792Z ls: cannot access /home/conda/bin: No such file or directory
```
Just FYI.

**Comment by pb-cdunn** [comment]
This will work for linux64. When we try to support OSX and Win, we will need to modify `nim.cfg` more carefully.

**Pull request merged**